### PR TITLE
ccgx: Reset the i2c bridge on status failure

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

**Comment by hughsie** [comment]
@superm1 this is how I see plugins being converted...

**Comment by hughsie** [comment]
@superm1 the other two look sane too?

File: plugins/nitrokey/fu-nitrokey-device.c [hunk 1]
```diff
@@ -13,26 +13,31 @@
 
 G_DEFINE_TYPE (FuNitrokeyDevice, fu_nitrokey_device, FU_TYPE_USB_DEVICE)
 
+typedef struct {
+	guint8		 command;
+	const guint8	*buf_in;
+	gsize		 buf_in_sz;
+	guint8		*buf_out;
+	gsize		 buf_out_sz;
+	GCancellable	*cancellable;
```
Comment (superm1) [inline plugins/nitrokey/fu-nitrokey-device.c]: I'm not sure plumbing cancellable all the way through makes sense.  It's `NULL` at it's earliest pass in.  Maybe just drop it as part of this?

**Comment by superm1** [comment]
Left a note about nitrokey.
VLI seems fine (although I will note that's a lot of sleep to go _before_ and _after_ which was already the case).

File: plugins/nitrokey/fu-nitrokey-device.c (hunk 1 above)
Comment (hughsie) [reply plugins/nitrokey/fu-nitrokey-device.c]: Fixed, thanks. I also noticed it was another SET_REPORT/GET_REPORT conversion, so I'll do that as a follow up.

**Comment by hughsie** [comment]
@superm1 what do you think of the last commit `fu_dell_dock_hub_recovery_impossible_cb`? I did ponder:

 * Allowing `fu_device_retry_add_recovery` to have a NULL `func` to signify that recovery is not possible
* Automatically adding `G_USB_DEVICE_ERROR_NO_DEVICE` in `FuUsbDevice` automatically for all devices.

Better ideas?

**Comment by superm1** [comment]
> @superm1 what do you think of the last commit `fu_dell_dock_hub_recovery_impossible_cb`? I did ponder:
> 
> * Allowing `fu_device_retry_add_recovery` to have a NULL `func` to signify that recovery is not possible

I like this idea.

> * Automatically adding `G_USB_DEVICE_ERROR_NO_DEVICE` in `FuUsbDevice` automatically for all devices.
> 
> Better ideas?

A simple wait callback like dell dock I wonder if we should have a callback error function for all plugins to use.  I would suspect this is going to be a common occurrence.

**Comment by hughsie** [comment]
> A simple wait callback like dell dock

How would we choose the timeout? Or something like `fu_device_retry_set_recovery_delay(FuDevice *d, guint timeout_ms)` that is used for all loops in `fu_device_retry()`

**Comment by hughsie** [comment]
> fu_device_retry_set_delay

This actually simplifies things quite a bit. Added, and rebased.

**Pull request merged**